client/app/components/alerts/alert-subscriptions.js

```diff
@@ -0,0 +1,166 @@
+        this.setState({
+          subscribers,
+          destinations,
+        });
+      });
+  }
+
+  state = {
+    selection: null,
+    subscribers: [],
+    destinations: [],
+  }
+
+  saveSubscriber = () => {
+    const sub = new this.props.AlertSubscription({ alert_id: this.props.alertId });
+    if (this.state.selection.value) {
+      sub.destination_id = this.state.selection.value;
+    }
+
+    sub.$save(
+      () => {
+        this.props.toastr.success('Subscribed.');
+        this.setState({
+          subscribers: this.state.subscribers.concat(sub),
+          destinations: this.state.destinations.filter(d => d.id !== this.state.selection.value),
+          selection: this.state.destinations && {
+            value: this.state.destinations[0].id,
+            label: this.state.destinations[0].name,
+            destination: this.state.destinations[0],
+          },
+        });
+      },
+      () => {
+        this.props.toastr.error('Failed saving subscription.');
+      },
+    );
+  }
+
+  unsubscribe = (subscriber) => {
+    const destination = subscriber.destination;
+    const user = subscriber.user;
+
+    subscriber.$delete(
+      () => {
+        this.props.toastr.success('Unsubscribed');
+        let destinations;
+        const subscribers = without(this.state.subscribers, subscriber);
+        if (destination) {
+          destinations = this.state.destinations.concat(destination);
+        } else if (user.id === this.props.currentUser.id) {
+          destinations = this.state.destinations.concat({
+            id: null,
+            name: this.props.currentUser.name + ' (Email)',
+            icon: 'fa-envelope',
+          });
+        }
+        const newState = { subscribers, destinations };
+        if (destinations.length === 1) {
+          newState.selection = {
+            value: destinations[0].id,
+            label: destinations[0].name,
+            destination: destinations[0],
+          };
+        }
+        this.setState(newState);
+      },
+      () => {
+        this.props.toastr.error('Failed unsubscribing.');
+      },
+    );
+  }
+
+  updateSelection = selection => this.setState({ selection })
+
+  render() {
+    return (
+      <div className="p-5">
+        <h4>Notifications</h4>
+        <div>
+          <Select
+            value={this.state.selection || null}
+            valueComponent={DestinationValue}
+            disabled={this.state.destinations.length === 0}
+            clearable={false}
+            options={this.state.destinations.map(d => ({ value: d.id, label: d.name, destination: d }))}
+            optionComponent={DestinationOption}
+
+            onChange={this.updateSelection}
+          />
+        </div>
+        <div className="m-t-5">
+          <button
+            className="btn btn-default"
+            onClick={this.saveSubscriber}
+            disabled={!this.state.selection || this.state.destinations.length === 0}
+            style={{ width: '50%' }}
```

Comment:
Not a fan of inline styles when they are not setting something computed at runtime.